akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -605,6 +605,14 @@ import scala.util.control.NonFatal
     // call has no effect and therefore does the right thing: nothing.
     try {
       inputs.foreach(_.onInternalError(reason))
+      // fail all sources
+      logics.foreach { logic ⇒
+        // source
+        if (logic.inCount == 0 && logic.outCount > 0) {
+          logic.failStage(reason)
+          interpreter.finalizeStage(logic)
```

Comment:
That might be a bug of `interpreter.finish` (which is only used from here). How do you prevent that `postStop` isn't called another time if it was already called before?